### DISCUSSION TITLE
signInWithMicrosoft unable to login

### DISCUSSION THREAD

**Discussion by xar** [capawesome-team/capacitor-firebase#216] [Q&A] [answered]
I'm trying to implement `signInWithMicrosoft` for my app. I have login handler which looks like this:

```
const result = await FirebaseAuthentication.signInWithMicrosoft({
    skipNativeAuth: false,
});

const credential = new OAuthProvider('microsoft.com').credential({
    idToken: result.credential?.idToken,
    accessToken: result.credential?.accessToken,
});

const userCredential = await signInWithCredential(auth, credential);
```

And I'm able to get `result.credential` which seems to be have all the required information. But after calling `signInWithCredential` I get `FirebaseError: Firebase: Invalid IdP response/credential:`

When I tried to use `skipNativeAuth: true` I didn't got any tokens into result.credential and was unable to proceed.

Is there any other specific steps needed when implementing Microsoft login. I tried to search in whole repo for some references.

**Comment by robingenz** [comment, marked as answer]
Duplicate of https://github.com/capawesome-team/capacitor-firebase/discussions/165.

>  As mentioned [here](https://cloud.google.com/identity-platform/docs/web/microsoft?hl=en), you can indeed not use the Microsoft sign-in with `signInWithCredential`.

--> Sign-in with Microsoft unfortunately only works with `skipNativeAuth=false` and without the Firebase JS SDK as it does not support `signInWithCredential`.

In the other discussion, another solution is proposed. But I could not test it yet.
